### DISCUSSION TITLE
The WHY Tree Style Tabs add-on (or similar add-on?) Can Not Be HORIZONTAL

### DISCUSSION THREAD

**Discussion by Ceyarrecks** [piroor/treestyletab#3633] [Q&A]
A simple (to my illiterate mind at least) question: 
**WHY** can not Tree Style Tabs, or any similar type add-on be _Horizontal_ as they once were?
  Back in the day, with version 0.19.20.x the option was there for any of the four sides, Top, Left, Right, Bottom;
  yet now.... only Right and Left? 
{seems almost akin to car purchase from Dealership: "oh, sorry, we only have one cylinder and 12 cylinder engine options."}
 I personally would **PAY**! (_handsomely_) as I know others would also, to have this ONE. OPTION. restored to be able to manage Firefox as was once so easily done,...  liken unto the Office Filing Cabinet Manila Folder Tabs we all grew up with and are obvious in their usefulness.
For a reminder, please review the attached picture showing the former Glory of 0.19.20.x of Tree Style Tabs as seen on Waterfox.
![(former Glory) of Tree Style Tabs](https://github.com/user-attachments/assets/985a1837-dbe1-483d-824a-5919a966d741)

**Comment by piroor** [comment]
Firefox 57 and later supports only WebExtensions addons, and Firefox does not allow WebExtensions addons to modify appearance of Firefox's native tabs. This is the reason why current TST (and other addons) provides only vertical tab bar in the sidebar.

Recent versions of Firefox supports vertical tab bar natively (it is in-progress), and the top tab bar goes away from the title bar area when you activate vertical tab bar. This will make it easy to addon authors providing custom horizontal tab bar rendered in the content area with no duplicated horizontal tab bars. (But sorry I have no idea to implement such an addon by me because I won't use it. I think someone seriously require such an addon should develop the addon.)

**Comment by Ceyarrecks** [comment]
Thank you!
A most thorough and detailed answer;
I understand your point of view of not being vested into what you would not use.

I also understand, that one whose expertise VASTLY outweighs my own (I have meager, rather, non-existent, programming experience)
and as I see the world organized in a specific manner,
 and with the changes abounding,
 are to me like taking Engineering Technical Drawings and blurring all the lines, all the text, and all the pictures of said technical drawing,
 leaving but a view of Abstract Expressionism. (e.g. https://artsology.com/blog/wp-content/uploads/2023/10/2023-07-11-2_Abstract_Art.jpg)

Additionally, *I* am highly motivated to compensate you for your time to develop said Tree Style Tabs for (either) LibreWolf or FireFox browser(s) that would endure.
Your expertise being familiar with the 'wilderness' that are Internet browsers would not be difficult to accomplish.

What say you?

I Thank You kindly for your time in both replying, but also, in kindly considering.

Hunt, Christopher Adam.


On Tue, 10 Dec 2024 21:31:22 -0800
piroor ***@***.***> wrote:

> Firefox 57 and later supports only WebExtensions addons, and Firefox does not allow WebExtensions addons to modify appearance of Firefox's native tabs. This is the reason why current TST (and other addons) provides only vertical tab bar in the sidebar.
> 
> Recent versions of Firefox supports vertical tab bar natively (it is in-progress), and the top tab bar goes away from the title bar area when you activate vertical tab bar. This will make it easy to addon authors providing custom horizontal tab bar rendered in the content area with no duplicated horizontal tab bars. (But sorry I have no idea to implement such an addon by me because I won't use it. I think someone seriously require such an addon should develop the addon.)
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/piroor/treestyletab/discussions/3633#discussioncomment-11528806
> You are receiving this because you authored the thread.
> 
> Message ID: ***@***.***>


-- 
Hunt, Christopher Adam
ハント、クリストファー・アダム
 亨特，克里斯托弗·亞當

*_The expectancy of the Strong_*
is to Protect & Support the weak, the wounded, the infirm.
Not miles ahead that the dangers they are being protected from are forgotten,
Not from behind that they are injured by said danger and friendly-fire,
but
Side. by. Side.
that they see What. they are being protected from,
How. they are being protected,
and to learn to protect others who are,...
weak,... wounded,... infirm,...
once they are become strong.

**Reply by piroor** [reply]
Ah, I meant that it is better for an addon (and most non-profit opensource software development projects) that a personal motivation, especially self-sufficient. Development based on justice or duty may let developers exhausted easily. On the other hand, self-sufficient development has strong motivation to continue it by the developer.

Long years ago I started addon development based on my very personal motivation. And I developed an all-in-one style extension which had very large number features based on requests from other people, but I couldn't continue the project because features I didn't use were breaking on every Firefox update and I couldn't spend enough time to maintain all features. And I had to shrink it to minimum features I continuously use.

At another point of view, I think that feature-oriented addon may be good first step to develop addons. A clear goal will reduce confusion and chaos from various feature requests.
